comit/src/lnd.rs

```diff
@@ -42,6 +42,20 @@ pub enum PaymentStatus {
     Failed,
 }
 
+#[derive(Copy, Clone, Debug, thiserror::Error)]
+#[error("payment statuses don't match: expected {expected} but received {received}")]
+pub struct StatusMismatch {
```

Comment:
```suggestion
pub struct PaymentStatusMismatch {
```
It's confusing to differentiate Payment and Invoice Mismatch error with `Status` vs `State`.

Reply:
Will refine this in a separate PR.

Reply:
https://github.com/comit-network/comit-rs/pull/2796